### PR TITLE
Prevent invalid beatmaps from being able to be selected on carousel

### PR DESCRIPTION
**Pull request by jai-x**
Attempt at fixing #863

### CODE REVIEW

File: osu.Game/Screens/Select/BeatmapDetails.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ public BeatmapInfo Beatmap
 
             set
             {
-                if (beatmap == value) return;
+                if (beatmap == value || value.OnlineBeatmapSetID < 1) return;
```
Comment (Jorolf) [inline osu.Game/Screens/Select/BeatmapDetails.cs]: what about beatmaps that aren't ranked? do they have an ID?

Comment (Tom94) [inline osu.Game/Screens/Select/BeatmapDetails.cs]: They do if they are uploaded, but we also need to support local beatmaps. Also, the beatmap set ID is unreliably. It's not always correct. So this is probably not a solution that'll work in the general case.

Comment (jai-sharma) [inline osu.Game/Screens/Select/BeatmapDetails.cs]: Would it be a better idea to prevent `BeatmapSets` being selected based on their hash values?

**Comment by peppy** [comment]
Not a correct solution. Did you even check how the main menu theme is being added? If you did you would understand there is a far better solution.

**Comment by jai-sharma** [comment]
Read your comment after submitting, ignore this and I'll try and do it properly

**Comment by peppy** [comment]
all you need to do is ensure `DeletePending` aren't displayed.

**Comment by peppy** [comment]
Did you test these changes? If you use the playlist control to select "circles" while at song select the song select display becomes out of sync with the loaded beatmap, causing badly defined behaviour.

How about just hiding deleted items from the playlist, like they should be

**Comment by jai-sharma** [comment]
I was under assuming that "Circles" should always have been selectable from the music controller if someone wanted to listen to it and that the select screen should ignore it if selected. I probably should have just mirrored behaviour in stable.

**Comment by peppy** [comment]
This is no longer required (and has been addressed int he recent fixes I've pushed).

**Pull request closed without merging**